src/terrain/terrain.cpp

```diff
@@ -918,15 +913,35 @@ Terrain::get_path(
             }
         }
     }
-    std::vector<const T*> out;
-    return out;
+    std::vector<const T*> path;
+    return path;
+}
+
+uint32_t
+Terrain::get_voxel(int x, int y, int z) const {
+    // using static ints to prevent dereferencing
+    static unsigned int get_voxel_mat_id = 0;
+    static unsigned int get_voxel_color_id = 0;
+    static uint32_t get_voxel_out_color = 0;
```

Comment:
Dereferencing? What??

I don't get what theses mean.
Also static integral values are always initialized to 0.
And why did you use uint32 and unsigned. Just used uint32

Reply:
dereferencing material

ascend to a height level of knowledge
ok sure, but reasons
because I'm just like that

So the point of this is to prevent some (%80+) of dereferencing. World::material is being dereferenced to find the color of the tile. The previous material id, and color id are cashed so that materials do not need to be referenced, and searched through. The reason that the values are explicitly initialized to 0 is because this hard-codes the color of "air" to 0. I would not want to implicitly do that.
the uint32 things is because if the computer has some other unsigned int (not 32) it probably does that because it is faster. That then is good for mat_id, and color_id, but the final color requires 32 bits.

Reply:
> So the point of this is to prevent some (%80+) of dereferencing. World::material is being dereferenced to find the color of the tile. The previous material id, and color id are cashed so that materials do not need to be referenced, and searched through.

Oh that makes sense. Maybe a better name would then be `previous_*_id`?

> The reason that the values are explicitly initialized to 0 is because this hard-codes the color of "air" to 0. I would not want to implicitly do that.

Probably should use a variable then. I'd recommend having an `enum` of material types.

> the uint32 things is because if the computer has some other unsigned int (not 32) it probably does that because it is faster. That then is good for mat_id, and color_id, but the final color requires 32 bits

Nope, its not because its faster. All additions/subtractions/bit shifts are sub-cycle operations for all variables smaller or equal to a machine word (pointer/unsigned long long). Machine word size is the bit-ness of a machine, so either 32 or 64 bits. The sizes for char/short/int/long/long long are chosen to be evenly distributed between char = 1B and long long = word size. There are very few 32-bit OSes now, almost all Linux and Windows installs and all MacOS installs are 64-bit, so `unsigned int` == `uint_32`. Just use `uint_32`, or if you don't need that much space, `uint_16`. You'll save some bytes and be 100% sure your variable will fit.